wordpress-mobile/WordPress-Android
# Plans on site domain display updated card

Fixes #19807, #19808, #19809   

Updates card with `upgrade to a plan` CTA, besides existing `get a domain` CTA on domains dashboard for those on free plan .

| Before | After |
|--------|--------|
| ![Screenshot_20231219_193130](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/1a127106-0ed5-495e-ba72-22b9e91294e8) |![Screenshot_20231220_112652](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/63988d4a-defd-4d28-b6ca-953d2470b362) | 

-----

## To Test:

Test 1
- Launch JP app
- `Select` a site with a free plan, have admin role
- `Go` to **Domains** (My site -> domains or My site -> More... -> domain)
- Verify a new card is displayed as in second image above
- `Tap` on **Upgrade to a plan**
- `Verify` it launches plans purchase flow like plans card on dashboard
- Return to domains
- `Tap` on **Just search for a domain**
- `Verify` it launches domain purchase flow as before

Test 2
- For completeness, also verify with a site that has a paid plan
- `Verify` it doesn't display the card above, but the card as before

Test 3
- `Verify` analytics tracker in AS like 
🔵 Tracked: domains_dashboard_get_plan_tapped, Properties: {"blog_id":196875124,"site_type":"blog","is_jetpack":false}
-----

## Regression Notes

1. Potential unintended areas of impact

    - None

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual, and unit tests

3. What automated tests I added (or what prevented me from doing so)

    - Updated unit tests

-----

## PR Submission Checklist:
